presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/TypeUtils.java

```diff
@@ -238,6 +262,19 @@ private static Object prestoNativeToJdbcObject(ConnectorSession session, Type pr
             return new Date(UTC.getMillisKeepLocal(DateTimeZone.getDefault(), millis));
         }
 
+        if (TIMESTAMP.equals(prestoType)) {
+            if (session.isLegacyTimestamp()) {
+                ZoneId sessionZone = ZoneId.of(session.getTimeZoneKey().getId());
+                return fromPrestoLegacyTimestamp((long) prestoNative, sessionZone);
+            }
+            return fromPrestoTimestamp((long) prestoNative);
+        }
+
+        if (TIMESTAMP_WITH_TIME_ZONE.equals(prestoType)) {
+            long millisUtc = unpackMillisUtc((long) prestoNative);
+            return new java.sql.Timestamp(millisUtc);
```

Comment:
`Timestamp` is already imported, you don't need FQN.

Reply:
Fixed